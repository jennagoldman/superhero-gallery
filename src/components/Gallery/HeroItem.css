.heroDiv {
  display: inline-block;
  width: 15rem;
}

.heroName {
  font-family: 'Bangers', sans-serif;
  font-weight: 700;
  letter-spacing: 0.1rem;
}

.heroDiv img {
  width: 15rem;
  height: 15rem;
  border-radius: 50%;
}

a {
  color: #007BFF;
  font-size: 0.75em;
  text-decoration: none;
  text-transform: uppercase;
}

a:hover {
  font-weight: 700;
}