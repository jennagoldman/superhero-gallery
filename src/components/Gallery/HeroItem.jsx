import React from 'react';
import styles from './HeroesItem.css';

const HeroItem = ({ name, imageUrl, comicsUrl }) => (
  <div className={styles.heroDiv}>
    <p className={styles.heroName}>{name}</p>
    <img src={imageUrl} alt={name} />
    <a href={comicsUrl} target="_blank" rel="noopener noreferrer">View Comics</a>
  </div>
);

export default HeroItem;
