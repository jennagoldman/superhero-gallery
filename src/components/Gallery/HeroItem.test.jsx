import React from 'react';
import { shallow } from 'enzyme';
import HeroItem from './HeroItem.jsx';

describe('ArtistsList', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<HeroItem name="Loki" imageUrl="http://i.annihil.us/u/prod/marvel/i/mg/d/90/526547f509313.jpg" comicsUrl="http://marvel.com/comics/characters/1009407/loki?utm_campaign=apiRef&utm_source=1f02de49106b672f61d699663164616b" />);
    expect(wrapper).toMatchSnapshot();
  });
});
