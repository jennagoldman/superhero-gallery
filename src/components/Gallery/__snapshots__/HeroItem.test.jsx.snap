// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistsList matches a snapshot 1`] = `
<div
  className="heroDiv"
>
  <p
    className="heroName"
  >
    Loki
  </p>
  <img
    alt="Loki"
    src="http://i.annihil.us/u/prod/marvel/i/mg/d/90/526547f509313.jpg"
  />
  <a
    href="http://marvel.com/comics/characters/1009407/loki?utm_campaign=apiRef&utm_source=1f02de49106b672f61d699663164616b"
    rel="noopener noreferrer"
    target="_blank"
  >
    View Comics
  </a>
</div>
`;
